src/App.js

```diff
@@ -23,21 +49,126 @@ class App extends React.Component {
     }));
   };
 
+  // check for winner using rank
+  // ask MS how to tidy this up
+  checkWinner = () => {
+    if (this.state.currCards.length > 1) {
+      if (this.state.currCards[0].rank > this.state.currCards[1].rank) {
+        this.setState({
+          winnerOfRound: 1,
+          numOfRounds: this.state.numOfRounds - 1,
+          playerOneScore: this.state.playerOneScore + 1,
+          message: "Player 1 wins this round!",
+        });
+      } else if (this.state.currCards[0].rank < this.state.currCards[1].rank) {
+        this.setState({
+          winnerOfRound: 2,
+          numOfRounds: this.state.numOfRounds - 1,
+          playerTwoScore: this.state.playerTwoScore + 1,
+          message: "Player 2 wins this round!",
+        });
+      } else {
+        this.setState({
+          winnerOfRound: 0,
+          numOfRounds: this.state.numOfRounds - 1,
+          message: "It's a tie!",
+        });
+      }
+    } else console.log("cards are not dealt yet.");
+  };
+
+  // react method takes two arguments
+  componentDidUpdate = (prevProp, prevState) => {
+    if (prevState.currCards !== this.state.currCards) {
+      this.checkWinner();
+    }
+  };
+
+  finalGameWinner = () => {
+    if (this.state.playerOneScore > this.state.playerTwoScore) {
+      return "Player 1 wins the game!";
+    } else if (this.state.playerOneScore < this.state.playerTwoScore) {
+      return "Player 2 wins the game!";
+    } else {
+      return "It's a game tie between player 1 & 2.";
+    }
+  };
+
+  resetGame = () => {
+    this.setState({
+      // Set default value of card deck to new shuffled deck
+      cardDeck: makeShuffledDeck(),
+      // currCards holds the cards from the current round
+      currCards: [],
+      winnerOfRound: 0,
+      message: "",
+      playerOneScore: 0,
+      playerTwoScore: 0,
+      numOfRounds: 26,
+    });
+  };
+
   render() {
-    const currCardElems = this.state.currCards.map(({ name, suit }) => (
-      // Give each list element a unique key
-      <div key={`${name}${suit}`}>
-        {name} of {suit}
-      </div>
-    ));
+    // followed along zoom recording to display cards
+    const currCardElems = this.state.currCards.map(
+      ({ name, suit, image, rank }, index) => {
+        console.log(suit);
```

Comment:
you should try to remove console.logs in your code submissions. they're meant to be for debugging only